stablehlo/dialect/Base.cpp

```diff
@@ -72,31 +72,34 @@ bool isCompatibleElementTypeForHloTypeInference(Type tp1, Type tp2) {
   tp1 = getElementTypeOrSelf(tp1);
   tp2 = getElementTypeOrSelf(tp2);
 
-  // Quantization: In the most general case, we allow any combination of
-  // quantized/non-quantized across any combination of operands/results,
-  // and some differences in quantization parameters across operands/results.
-  // Individual ops may introduce additional constraints.
   auto qtp1 = tp1.dyn_cast<quant::QuantizedType>();
   auto qtp2 = tp2.dyn_cast<quant::QuantizedType>();
+
+  if (!qtp1 && !qtp2) {
+    // Both are non quantized
+    return tp1 == tp2;
+  }
   if (qtp1 && qtp2) {
+    // Both are quantized
     if (qtp1.getStorageType() != qtp2.getStorageType() ||
         qtp1.getStorageTypeMin() != qtp2.getStorageTypeMin() ||
-        qtp1.getStorageTypeMax() != qtp2.getStorageTypeMax())
+        qtp1.getStorageTypeMax() != qtp2.getStorageTypeMax() ||
+        qtp1.getExpressedType() != qtp2.getExpressedType()) {
       return false;
+    }
+
+    auto qpatp1 = qtp1.dyn_cast<quant::UniformQuantizedPerAxisType>();
+    auto qpatp2 = qtp2.dyn_cast<quant::UniformQuantizedPerAxisType>();
+    if (qpatp1 && qpatp2) {
```

Comment:
replace
```
    if (qpatp1 && qpatp2) {
      // Both are also per-axis quantized
      // Don't match dimensions, Per OP verifier will do it.
      return true;
    }
    // return true if both are per-tensor quantized
    return !(qpatp1 || qpatp2);
``` 
with 
```
return qpatp1 == qpatp2;
```

Reply:
qpatp* are ptr.. `return qpatp1 == qpatp2;` won't return desired value.
changed it to `return ((qpatp1 && qpatp2) || (!qpatp1 && !qpatp2));` thank you.